# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/jimmy/mplayer/linux-asm-byteorder_h.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -dur MPlayer-1.0pre5/libmpdemux/cddb.c MPlayer-1.0pre5-p/libmpdemux/cddb.c
--- MPlayer-1.0pre5/libmpdemux/cddb.c	2004-04-26 11:33:35.000000000 +0200
+++ MPlayer-1.0pre5-p/libmpdemux/cddb.c	2004-11-06 05:57:35.118682456 +0100
@@ -11,6 +11,17 @@
  *	 Code release under GPL
  *
  */
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
 
 #include "config.h"
 
diff -dur MPlayer-1.0pre5/libmpdemux/stream_vcd.c MPlayer-1.0pre5-p/libmpdemux/stream_vcd.c
--- MPlayer-1.0pre5/libmpdemux/stream_vcd.c	2004-07-12 00:23:57.000000000 +0200
+++ MPlayer-1.0pre5-p/libmpdemux/stream_vcd.c	2004-11-06 05:55:26.123292736 +0100
@@ -1,4 +1,16 @@
 
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
 #include "config.h"
 
 #ifdef HAVE_VCD
--- MPlayer-1.0pre5/osdep/kerneltwosix.h	2004-01-10 10:48:15.000000000 +0100
+++ MPlayer-1.0pre5-p/osdep/kerneltwosix.h	2004-11-06 06:57:15.776339752 +0100
@@ -1,5 +1,20 @@
 #include <linux/version.h>
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,70)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,9)
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+#define __le16 __bitwise __u16
+#define __be16 __bitwise __u16
+#define __le32 __bitwise __u32
+#define __be32 __bitwise __u32
+#define __le64 __bitwise __u64
+#define __be64 __bitwise __u64
+
+#endif
 #define __KERNEL__
 #include <linux/thread_info.h>
 #include <linux/list.h>
